tests/bootstrap.js

```diff
@@ -0,0 +1,241 @@
+ *     uncompressed mode.
+ *
+ *     Load this file in a <script> tag in a web page to use the
+ *     Closure Library debug module loader to load Blockly in
+ *     uncompressed mode.
+ *
+ *     You must use a <script> tag to load this script first, then
+ *     import bootstrap_done.mjs in a <script type=module> to wait for
+ *     bootstrapping to finish.
+ *
+ *     See tests/playground.html for example usage.
+ *
+ *     Exception: for speed and compatibility reasons, if this is
+ *     script is loaded in Internet Explorer or from a host other than
+ *     localhost, blockly_compressed.js (et al.) will be loaded
+ *     instead.  IE 11 doesn't understand modern JavaScript, and
+ *     because of the sequential module loading carried out by the
+ *
+ *     (Note also that this file eschews certain modern JS constructs,
+ *     like template literals, for compatibility with IE 11.)
+ *
+ *     The bootstrap code will consult a BLOCKLY_BOOTSTRAP_OPTIONS
+ *     global variable to determine what to load.  See further
+ *     documentation inline.
+ *
+ */
+'use strict';
+
+(function() {
+  // Values usedd to compute default bootstrap options.
+  const isIe = navigator.userAgent.indexOf('MSIE') !== -1 ||
+      navigator.appVersion.indexOf('Trident/') > -1;
+  const localhosts = ['localhost', '127.0.0.1', '[::1]'];
+  const isLocalhost = localhosts.includes(location.hostname);
+  const isFileUrl = location.origin === 'file://';
+
+  // Default bootstrap options.
+  const options = {
+    // Decide whether to use compmiled mode or not.  Please see issue
+    // #5557 for more information.
+    loadCompressed: isIe || !(isLocalhost || isFileUrl),
+
+    // URL of the blockly repository.  This is needed for a few reasons:
+    //
+    // - We need an absolute path instead of relative path because the
+    //   advanced_playground the and regular playground are in
+    //   different folders.
+    // - We need to get the root directory for blockly because it is
+    //   different for github.io, appspot and local.
+    //
+    // Default value will work so long as top-level page is loaded
+    // from somewhere in tests/.
+    root: window.location.href.replace(/\/tests\/.*$/, '/'),
+
+    // List of deps files to load.  Paths relative to root.
+    depsFiles: [
+      'build/deps.js',
+    ],
+    
+    // List of goog.modules to goog.require.
+    requires: [
+      'Blockly',
+      'Blockly.libraryBlocks',
+      'Blockly.Dart.all',
+      'Blockly.JavaScript.all',
+      'Blockly.Lua.all',
+      'Blockly.PHP.all',
+      'Blockly.Python.all',
+    ],
+
+    // List of scripts to load in compressed mode, instead of
+    // requires.  Paths relative to root.
+    compressedScripts: [
+      'build/blockly_compressed.js',
+      'build/blocks_compressed.js',
+      'build/dart_compressed.js',
+      'build/javascript_compressed.js',
+      'build/lua_compressed.js',
+      'build/php_compressed.js',
+      'build/python_compressed.js',
+    ],
```

Comment:
Could you provide some more info about how this will work when hosting? My understanding was that the build directory wasn't going to be checked in. I assume we can just copy everything into the uploaded directory when hosting on App Engine, but what if we want to use Github Pages?